pkg/server/server.go

```diff
@@ -349,7 +349,8 @@ func (s *Server) Check(ctx context.Context, req *openfgapb.CheckRequest) (*openf
 	res := &openfgapb.CheckResponse{
 		Allowed: resp.Allowed,
 	}
-
+	s.transport.SetHeader(ctx, "db-reads", strconv.Itoa(int(dbReads)))
```

Comment:
This should be put behind a configuration IMO.

If `--debug-trace` or something is enabled then these sub-specs of the API are enabled.

Reply:
This should honor the `trace` in the query params, rather than the `debug-trace` flag

Also, can we change the header to:
- `X-OpenFga-Query-ReadCount`
- or `OpenFga-Query-ReadCount`

Reply:
Query-Read seems a bit redundant.  Thus, I would recommend choosing one or the other.
Also, we should make the name a constant as it may be used later.